tgui/packages/tgui/interfaces/PaperSheet.js

```diff
@@ -0,0 +1,153 @@
+/**
+ * @file
+ * @copyright 2020
+ * @author Original Warlockd (https://github.com/warlockd)
+ * @author Changes stylemistake
+ * @license MIT
+ */
+
+import { Tabs, Box, Flex, Button, TextArea } from '../components';
+import { useBackend, useSharedState, useLocalState } from '../backend';
+import { Window } from '../layouts';
+// import marked from 'marked';
+import marked from 'marked';
+import DOMPurify from 'dompurify';
+// There is a sanatize option in marked but they say its deprecated.
+// Might as well use a proper one then
+
+import { createLogger } from '../logging';
+import { Fragment } from 'inferno';
+
+const logger = createLogger('PaperSheet');
+
+const run_marked_default = value => {
+  const sanitizer = DOMPurify.sanitize;
+  // too much?
+  // return sanitizer(marked(sanitizer(value),
+  //   { breaks: true, smartypants: true });
+  return sanitizer(marked(value,
+    { breaks: true, smartypants: true }));
+};
+
+const PaperSheetView = (props, context) => {
+  const { data } = useBackend(context);
+  const {
+    paper_color = "white",
+    pen_color = "black",
+    text = '',
+  } = data;
+  const {
+    value = text || '',
+    ...rest
+  } = props;
+  // We use this for caching so we don't keep refreshing it each time
+  const [marked_text, setMarkedText] = useLocalState(context, 'marked_text',
+    { __html: run_marked_default(value) });
+  return (
+    <Box
+      backgroundColor={paper_color}
+      color={pen_color}
+      {...rest}
+      dangerouslySetInnerHTML={marked_text} />
+  );
+};
+
+const PaperSheetEdit = (props, context) => {
+  const { act, data } = useBackend(context);
+  const [text, setText] = useLocalState(context, 'text', data.text || '');
+  const [marked_text, setMarkedText] = useLocalState(context, 'marked_text',
+    { __html: run_marked_default(text) });
+  const [
+    previewSelected,
+    setPreviewSelected,
+  ] = useLocalState(context, 'preview', "Preview");
+  const {
+    paper_color = "white",
+    pen_color = "black",
+  } = data;
+  const onInputHandler = (e, value) => {
+    if (value.length < 1000) {
```

Comment:
We're going to need server side checks too.

The whole reason paper was removed is due to someone acting maliciously.  Front end checks aren't going to stop that.

Reply:
I was afraid of that.  Checking for javascript or bad html won't be a problem,  The issue I have been having is if you send like a 100k text file, it kills byond.  Is there a way to stop it from that side?

Reply:
That is, if someone used a javascript injection attack, it would do it.  But then you could do the same for any ui_act updates.

Reply:
Take me an hour or so to find the functions but I will get some back end stuff to kill incoming html

Reply:
it should just check length, what the content is shouldn't matter, since it's only going to be rendered on the clientside again

Reply:
Thats simple enough to do.  I will fix post haste.